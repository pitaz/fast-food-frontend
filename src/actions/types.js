export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const SIGN_UP_ERRORS = 'SIGN_UP_ERRORS';
export const REMOVE_ERROR_MESSAGE = 'REMOVE_ERROR_MESSAGE';
export const SET_CURRENT_USER_FAIL = 'SET_CURRENT_USER_FAIL';
export const LOG_OUT_USER = 'LOG_OUT_USER';
export const GET_MEALS = 'GET_MEALS ';
