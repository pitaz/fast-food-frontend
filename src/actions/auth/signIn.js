import axios from "axios";
import jwt from "jsonwebtoken";
import Cookie from "cookies-js";
import toastr from "toastr";
import {
  SET_CURRENT_USER,
  SIGN_UP_ERRORS,
	REMOVE_ERROR_MESSAGE,
	SET_CURRENT_USER_FAIL
} from "../types";

export const setCurrentUser = user => ({
  type: SET_CURRENT_USER,
  user
});

export const setCurrentUserError = error => ({
  type: SET_CURRENT_USER_FAIL,
  error
});

export const removeErrorMsgCompleted = user => ({
  type: REMOVE_ERROR_MESSAGE,
  user
});

export const removeErrorMsg = () => (dispatch) => {
 dispatch(removeErrorMsgCompleted());
};

const signIn = data => (dispatch) => {
	return axios.post('https://fast-food-pitaz.herokuapp.com/api/v1/auth/login', data).then(
		(res) => {
			const { token } = res.data.data;
			const { message } = res.data;
			Cookie.set('token', token);
			toastr.success(message);
			dispatch(setCurrentUser(jwt.decode(token)));
			return true;
		}
	).catch((error) => {
		dispatch(setCurrentUserError(error.response.data));
	});
};

export default signIn;
