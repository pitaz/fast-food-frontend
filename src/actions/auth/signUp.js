import axios from "axios";
import jwt from "jsonwebtoken";
import Cookie from "cookies-js";
import toastr from "toastr";
import {
  SET_CURRENT_USER,
  SIGN_UP_ERRORS,
	REMOVE_ERROR_MESSAGE,
	SET_CURRENT_USER_FAIL
} from "../types";

export const setCurrent = user => ({
  type: SET_CURRENT_USER,
  user
});

export const setCurrentUserError = error => ({
  type: SET_CURRENT_USER_FAIL,
  error
});

export const registrationError = error => ({
  type: SIGN_UP_ERRORS,
  error
});

export const removeErrorMsgCompleted = user => ({
  type: REMOVE_ERROR_MESSAGE,
  user
});

export const removeErrorMsg = () => (dispatch) => {
 dispatch(removeErrorMsgCompleted());
};

const signUp = data => (dispatch) => {
	return axios.post('https://fast-food-pitaz.herokuapp.com/api/v1/auth/signup', data).then(
		(res) => {
			const { token } = res.data.data;
			const { message } = res.data;
			Cookie.set('token', token);
			toastr.success(message);
			dispatch(setCurrent(jwt.decode(token)));
			return true;
		}
	).catch((error) => {
		dispatch(setCurrentUserError(error.response.data));
	});
};

export default signUp;
