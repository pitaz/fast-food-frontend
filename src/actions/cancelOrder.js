import axios from 'axios';
import {toastr} from 'react-redux-toastr';
import * as types from './types';

export const cancelOrderSuccess = orderId => ({
  type: types.CANCEL_ORDER,
  orderId,
});
export const cancelOrderError = error => ({
  type: types.CANCEL_ORDER_FAIL,
  error,
});

const  cancelOrder = (orderId) => (dispatch) => {
	return axios.delete(`https://fast-food-pitaz.herokuapp.com/api/v1/orders/${orderId}`).then(
		(res) => {
      dispatch(cancelOrderSuccess(orderId));
      toastr.warning('Order has been cancelled');
		}
	).catch((error) => {
		dispatch(cancelOrderError(error.response));
	});
};

export default cancelOrder;
