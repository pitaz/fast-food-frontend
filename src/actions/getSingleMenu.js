import axios from 'axios';
import * as types from './types';

export const getSingleMenu = menu => ({
  type: types.GET_SINGLE_MENU,
  menu,
});
export const getSingleMenuError = errors => ({
  type: types.GET_SINGLE_MENU_FAIL,
  errors,
});

const fetchSingleMenu = (mealId) => (dispatch) => {
	return axios.get(`https://fast-food-pitaz.herokuapp.com/api/v1/menu/${mealId}`).then(
		(res) => {
      dispatch(getSingleMenu(res.data.data.item));
		}
	).catch((error) => {
		dispatch(getSingleMenuError(error.response.data));
	});
};

export default fetchSingleMenu;
