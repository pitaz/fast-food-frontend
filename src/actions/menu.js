import axios from 'axios';
import * as types from './types';

export const getMeals = menu => ({
  type: types.GET_MEALS,
  menu,
});
export const getMealsError = error => ({
  type: types.GET_MEALS,
  error,
});

const fetchMeal = () => (dispatch) => {
	return axios.get('https://fast-food-pitaz.herokuapp.com/api/v1/menu').then(
		(res) => {
      dispatch(getMeals(res.data.data.items));
		}
	).catch((error) => {
		console.log(error);
		
		dispatch(getMealsError(error.response.data));
	});
};

export default fetchMeal;
