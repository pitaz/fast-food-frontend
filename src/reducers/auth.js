import isEmpty from "is-empty";
import {
  SET_CURRENT_USER,
  SIGN_UP_ERRORS,
  REMOVE_ERROR_MESSAGE,
  SET_CURRENT_USER_FAIL,
	LOG_OUT_USER
} from "../actions/types";

const initialState = {
  isAuthenticated: false,
  user: {},
  error: {}
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_CURRENT_USER:
      return {
        ...state,
        isAuthenticated: !isEmpty(action.user),
        user: action.user
      };

    case SET_CURRENT_USER_FAIL:
      return {
				...state,
				isAuthenticated: false,
				user: {},
				error: action.error
			};

			case SIGN_UP_ERRORS:{
      return {
				...state,
				isAuthenticated: false,
				error: action.error
			};
		}
			case REMOVE_ERROR_MESSAGE:
      return {
				error: {}
			};

			case LOG_OUT_USER:
			return {
				...state,
				...initialState
			};
			default: return state;
  }
};
