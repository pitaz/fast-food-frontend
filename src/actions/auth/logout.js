import Cookie from 'cookies-js';
import { LOG_OUT_USER } from '../../actions/types';

export const removeCurrentUser = () => ({
 type: LOG_OUT_USER
});

const logout = () => (dispatch) => {
 Cookie.expire('token');
 dispatch(removeCurrentUser({}));
};

export default logout;
