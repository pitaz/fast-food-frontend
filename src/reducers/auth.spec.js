/* eslint-disable no-undef */
import expect from 'expect';
import * as types from '../actions/types';
import auth from './auth';

describe('select_reducer', () => {
  it('renders with initial state', () => {
    const action = { type: types.SET_CURRENT_USER };
    const initialState = { isAuthenticated: false, user: undefined };
    expect(auth({}, action)).toEqual(initialState);
  });

  it('renders correctly', () => {
    const action = { type: types.SET_CURRENT_USER };
    expect(auth(undefined, action)).toMatchSnapshot();
  });

  it('returns the correct state', () => {
    const action = { type: types.SET_CURRENT_USER, user: {} };
    expect(auth(undefined, action)).toMatchSnapshot();
  });

  it('should set user if user is succesfully stored', () => {
    const initialState = {
      isAuthenticated: false,
      user: {},
      error: {},
    };

    const state = auth(initialState, {
      type: types.SET_CURRENT_USER,
      isAuthenticated: true,
      user: {
        userId: 1,
        email: 'email',
        username: 'peter',
        image: 'https://hardwaremassive.com/s…cture.jpg',
        roleId: 1,
      },
      error: {},
    });
    expect(state).toEqual({
      isAuthenticated: true,
      user: {
        userId: 1,
        email: 'email',
        username: 'peter',
        image: 'https://hardwaremassive.com/s…cture.jpg',
        roleId: 1,
      },
      error: {},
    });
  });

  it('renders correctly', () => {
    const action = { type: types.SET_CURRENT_USER_FAIL };
    expect(auth(undefined, action)).toMatchSnapshot();
  });

  it('returns the correct state', () => {
    const action = { type: types.SET_CURRENT_USER, error: {} };
    expect(auth(undefined, action)).toMatchSnapshot();
  });

  it('should set user if user login was unsuccessful', () => {
    const initialState = {
      isAuthenticated: false,
      user: {},
      error: {},
    };

    const state = auth(initialState, {
      type: types.SET_CURRENT_USER_FAIL,
      isAuthenticated: false,
      user: {},
      error: { message: 'Login unsuccessful' },
    });
    expect(state).toEqual({
      isAuthenticated: false,
      user: {},
      error: { message: 'Login unsuccessful' },
    });
  });

  it('should return errors if login was unsuccessful', () => {
    const initialState = {
      isAuthenticated: false,
      user: {},
      error: {
        message: 'login unsuccessful'
      },
    };

    const state = auth(initialState, {
      type: types.SET_CURRENT_USER_FAIL,
      isAuthenticated: false,
      user: {},
      error: { message: 'Login unsuccessful' },
    });
    expect(state).toEqual({
      isAuthenticated: false,
      user: {},
      error: { message: 'Login unsuccessful' },
    });
  });
  it('returns the correct state', () => {
    const action = { type: types.LOG_OUT_USER };
    expect(auth(undefined, action)).toMatchSnapshot();
  });
});
