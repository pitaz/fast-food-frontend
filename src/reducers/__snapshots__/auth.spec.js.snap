// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_reducer renders correctly 1`] = `
Object {
  "error": Object {},
  "isAuthenticated": false,
  "user": undefined,
}
`;

exports[`select_reducer renders correctly 2`] = `
Object {
  "error": undefined,
  "isAuthenticated": false,
  "user": Object {},
}
`;

exports[`select_reducer returns the correct state 1`] = `
Object {
  "error": Object {},
  "isAuthenticated": false,
  "user": Object {},
}
`;

exports[`select_reducer returns the correct state 2`] = `
Object {
  "error": Object {},
  "isAuthenticated": false,
  "user": undefined,
}
`;
